testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -7714,6 +7714,277 @@ void assertSubFunReturns(boolean binary, String s, int start,
     }
   }
 
+  @Test void testFormatNumber() {
+    final SqlOperatorFixture f0 = fixture().setFor(SqlLibraryOperators.FORMAT_NUMBER);
+    f0.checkFails("^format_number(123, 2)^",
+        "No match found for function signature FORMAT_NUMBER\\(<NUMERIC>, <NUMERIC>\\)",
+        false);
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      // test with tinyint type
+      f.checkString("format_number(cast(0 as tinyint), 0)", "0",
+          "VARCHAR NOT NULL");
+      f.checkString("format_number(cast(1 as tinyint), 4)", "1.0000",
+          "VARCHAR NOT NULL");
+      f.checkString("format_number(cast(-1 as tinyint), 4)", "-1.0000",
+          "VARCHAR NOT NULL");
+      f.checkString("format_number(cast(12 as tinyint), 2)", "12.00",
+          "VARCHAR NOT NULL");
+      f.checkString("format_number(cast(123 as tinyint), 5)", "123.00000",
+          "VARCHAR NOT NULL");
+      f.checkString("format_number(cast(1 as tinyint), '#,###,###,###,###,###,##0.0000')",
```

Comment:
I think we could make the testing here a bit more concise, if you check out the `testAbsFunc` test or some others, I don't think adding the same few tests for each type is adding much value, maybe just one per type or something just to verify they're all compatible. What do you think?

Reply:
You are right. I will reduce the test cases for each types. And a reasonable way is to split the test cases into there general types: bigint/double/decimal, and add more test cases to cover. For other types, like tinyint/smallint/int, just add one test case is enough.